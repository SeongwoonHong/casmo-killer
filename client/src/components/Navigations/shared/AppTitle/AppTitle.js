import React from 'react';
import { Link } from 'react-router-dom';

import logo from '@assets/logo.svg';

import './AppTitle.scss';

const AppTitle = () => {

  return (
    <h1 className="App-title">
      <Link to="/" className="App-title__link">
        <img
          src={ logo }
          alt="Damso Logo"
          className="App-title__img"
        />
      </Link>
    </h1>
  );

};

export default AppTitle;
