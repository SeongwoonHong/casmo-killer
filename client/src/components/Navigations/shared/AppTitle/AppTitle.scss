.App-title {
    margin: 0;
    text-align: center;
    width: $main-menu-width - $top-nav-size;
    &__link {
        color: #fff;
        display: block;
        font-size: 22px;
        line-height: $top-nav-size;
        margin: 0 auto;
        width: 60%;
    }
    &__img {
        vertical-align: middle;
    }
    @media screen and (min-width: 992px) {
        &__link {
            font-size: 28px;
            margin-left: 0;
            text-shadow: 1px 1px 2px rgba(0, 0, 0, 1);
        }
    }
}
