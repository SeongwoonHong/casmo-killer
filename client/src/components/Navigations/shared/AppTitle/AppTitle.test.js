import React from 'react';
import { shallow } from 'enzyme';

import AppTitle from './index';

describe('<AppTitle />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(<AppTitle />);
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
