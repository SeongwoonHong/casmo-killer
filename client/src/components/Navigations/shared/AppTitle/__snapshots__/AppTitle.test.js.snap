// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTitle /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppTitle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="App-title__link"
        replace={false}
        to="/"
>
        <img
                alt="Damso Logo"
                className="App-title__img"
                src="logo.svg"
        />
</Link>,
      "className": "App-title",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <img
          alt="Damso Logo"
          className="App-title__img"
          src="logo.svg"
/>,
        "className": "App-title__link",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Damso Logo",
          "className": "App-title__img",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="App-title__link"
          replace={false}
          to="/"
>
          <img
                    alt="Damso Logo"
                    className="App-title__img"
                    src="logo.svg"
          />
</Link>,
        "className": "App-title",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="Damso Logo"
            className="App-title__img"
            src="logo.svg"
/>,
          "className": "App-title__link",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Damso Logo",
            "className": "App-title__img",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
